torchvision/transforms/v2/functional/_utils.py

```diff
@@ -43,28 +45,62 @@ def _name_to_dispatcher(name):
     try:
         return getattr(torchvision.transforms.v2.functional, name)
     except AttributeError:
-        raise ValueError(f"Could not find dispatcher with name '{name}'.") from None
+        raise ValueError(
+            f"Could not find dispatcher with name '{name}' in torchvision.transforms.v2.functional."
+        ) from None
 
 
 def register_kernel(dispatcher, datapoint_cls):
     if isinstance(dispatcher, str):
         dispatcher = _name_to_dispatcher(name=dispatcher)
+    elif not (
+        callable(dispatcher)
+        and getattr(dispatcher, "__module__", "").startswith("torchvision.transforms.v2.functional")
+    ):
+        raise ValueError(
+            f"Kernels can only be registered on dispatchers from the torchvision.transforms.v2.functional namespace, "
+            f"but got {dispatcher}."
+        )
+
+    if not (
+        isinstance(datapoint_cls, type)
+        and issubclass(datapoint_cls, datapoints.Datapoint)
+        and datapoint_cls is not datapoints.Datapoint
+    ):
+        raise ValueError(
+            f"Kernels can only be registered for subclasses of torchvision.datapoints.Datapoint, "
+            f"but got {datapoint_cls}."
+        )
+
     return _register_kernel_internal(dispatcher, datapoint_cls, datapoint_wrapper=False)
 
 
-def _get_kernel(dispatcher, datapoint_cls):
+def _get_kernel(dispatcher, input_type):
     registry = _KERNEL_REGISTRY.get(dispatcher)
     if not registry:
-        raise ValueError(f"No kernel registered for dispatcher '{dispatcher.__name__}'.")
+        raise ValueError(f"No kernel registered for dispatcher {dispatcher.__name__}.")
 
-    if datapoint_cls in registry:
-        return registry[datapoint_cls]
+    # in case we have an exact type match, we take a shortcut
+    if input_type in registry:
+        return registry[input_type]
 
-    for registered_cls, kernel in registry.items():
-        if issubclass(datapoint_cls, registered_cls):
-            return kernel
+    # in case of datapoints, we check if we have a kernel for a superclass registered
+    if issubclass(input_type, datapoints.Datapoint):
+        for cls in input_type.__mro__:
+            if cls is datapoints.Datapoint:
```

Comment:
A short comment to explain why this is here could be helpful (it took us a bit of time to decide on this so this is really not trivial)

```suggestion
            if cls is datapoints.Datapoint:
                # we don't want user-defined datapoints to dispatch to the pure Tensor kernels,
                # so we expliclty stop the mro traversal before hitting Tensor. We can even stop at Datapoint
                # since we don't allow kernels to be registered for Datapoints anyway.
```